src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -0,0 +1,191 @@
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.language.v1.EntityMention;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.LanguageServiceSettings;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/*
+* This class creates objects that analyse text by 
+* extracting the categories, the mood, the events.
+* The scope is to get a final set of the key words
+* from the text. 
+*/
+
+public final class TextAnalyser {
+  private final String message;
+  private static final double EPSILON = 0.00001;
+
+  public TextAnalyser(String message) {
+     this.message = message.toLowerCase();
+  }
+
+  public final LanguageServiceSettings getSettings() throws IOException {
+    HeaderProvider headerProvider =
+            FixedHeaderProvider.create("X-Goog-User-Project","google.com:gpostcard");
+    return LanguageServiceSettings.newBuilder()
+                .setHeaderProvider(headerProvider)
+                .build();
+  }
+
+  public float getSentimentScore() throws IOException {
+    try (LanguageServiceClient languageService = LanguageServiceClient.create(getSettings())) {
+      Document doc =
+        Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+        
+      return sentiment.getScore();
+    }
+  }
+
+  public String getMood() throws IOException {
+    float score = getSentimentScore();
+
+    if (Math.abs(score - 1) < EPSILON) {
+      return "very happy";
+    }
+ 
+    if (Math.abs(score + 1) < EPSILON) {
+      return "so pessimistic";
+    }
+ 
+    int position = 0;
+    String[] moods = new String[]{"neutral", "calm", "relaxed", "serene", "contented",
+                                  "joyful", "happy", "delighted", "excited", "thrilled",
+                                  "tense", "nervous", "stressed", "upset", "sad", 
+                                  "depressed", "bored", "fatigued", "pessimisctic"};
+ 
+    score = score * 10;
+    position = (int) score;
+ 
+    if (position < 0) {
+      position = position * (-1) + 9;
+    }
+ 
+    return moods[position];
+  }
+
+  private ClassifyTextResponse classify() throws IOException {
+    try (LanguageServiceClient language = LanguageServiceClient.create(getSettings())) {
+      Document document = 
+            Document.newBuilder().setContent(message).setType(Type.PLAIN_TEXT).build();
+      ClassifyTextRequest request = ClassifyTextRequest.newBuilder().setDocument(document).build();
+      
+      return language.classifyText(request);
+    } catch (Exception e) {
+      System.err.println("Not enough tokens (words) to actually get a category.");
```

Comment:
In future maybe use Logging (https://cloud.google.com/appengine/docs/standard/java/logs) which provides a better experience for engineers browsing and analysing the logs.